cicecore/cicedyn/general/ice_init.F90

```diff
@@ -1232,6 +1259,111 @@ subroutine input_data
+               write(nu_diag,*) subname//' WARNING: history_format='//trim(history_format)// &
+                                ' is deprecated, please update namelist settings'
+            endif
+            history_format = 'pnetcdf2'
+         else
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' ERROR: lcdf64 is T and history_format not supported for '//trim(history_format)
+            endif
+            abort_list = trim(abort_list)//":52"
+         endif
+
+         if (restart_format == 'default' .or. restart_format == 'pio_netcdf') then
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' WARNING: restart_format='//trim(restart_format)// &
+                                ' is deprecated, please update namelist settings'
+            endif
+            restart_format = 'cdf2'
+         elseif (restart_format == 'pio_pnetcdf') then
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' WARNING: restart_format='//trim(restart_format)// &
+                                ' is deprecated, please update namelist settings'
+            endif
+            restart_format = 'pnetcdf2'
+         else
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' ERROR: lcdf64 is T and restart_format not supported for '//trim(restart_format)
+            endif
+            abort_list = trim(abort_list)//":53"
+         endif
+      else
+         if (history_format == 'default' .or. history_format == 'pio_netcdf') then
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' WARNING: history_format='//trim(history_format)// &
+                                ' is deprecated, please update namelist settings'
+            endif
+            history_format = 'cdf1'
+         elseif (history_format == 'pio_pnetcdf') then
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' WARNING: history_format='//trim(history_format)// &
+                                ' is deprecated, please update namelist settings'
+            endif
+            history_format = 'pnetcdf1'
+         endif
+
+         if (restart_format == 'default' .or. restart_format == 'pio_netcdf') then
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' WARNING: restart_format='//trim(restart_format)// &
+                                ' is deprecated, please update namelist settings'
+            endif
+            restart_format = 'cdf1'
+         elseif (restart_format == 'pio_pnetcdf') then
+            if (my_task == master_task) then
+               write(nu_diag,*) subname//' WARNING: restart_format='//trim(restart_format)// &
+                                ' is deprecated, please update namelist settings'
+            endif
+            restart_format = 'pnetcdf1'
+         endif
+      endif
+
       if (ktransport <= 0) then
```

Comment:
It's happening later in the code.  If a format is selected that isn't compatible with the ICE_IOTYPE, it will abort when it's used in the io code, not during initialization.  We could add something in the initialization, but that would require adding some new CPPs that trigger from the ICE_IOTYPE, and I'm reluctant to do that just to move an abort earlier.